tools/webhook_test.py

```diff
@@ -125,6 +147,76 @@ def int_or_default(input_parm):
         raw_input(), payload["message"][input_parm])
 
 
+def get_gym_info(gym_id):
+    """ Gets the information about the gym. """
+    return _gym_info.get(gym_id, _default_gym_info)
+
+
+def check_gym(questionable_input):
+    if get_gym_info(questionable_input)['name'] != "unknown":
+        print "Gym found = {}".format(get_gym_info(questionable_input)['name'])
+        return questionable_input, get_gym_info(questionable_input)['name']
+    else:
+        print "Not a valid gym. Please try again..\n>"
+        check_gym(raw_input())
+
+
+def gym_or_invalid(prm, prm2):
+    payload["message"][prm], payload["message"][prm2] = check_gym(raw_input())
```

Comment:
When I run this and get to the `Enter gym id for raid` prompt, and I put in an empty string once, then a valid gym id the next time, this errors out.
It appears this is because of the recursive calls of `check_gym`. It looks like that it returns the data accurately once, and gives invalid data the other times.
```
Gym found = The Battle of Carthage
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2017.2.4\helpers\pydev\pydevd.py", line 1599, in <module>
    globals = debugger.run(setup['file'], None, None, is_module)
  File "C:\Program Files\JetBrains\PyCharm 2017.2.4\helpers\pydev\pydevd.py", line 1026, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "C:/PycharmProjects/mali/tools/webhook_test.py", line 444, in <module>
    gym_or_invalid("gym_id", "gym_name")
  File "C:/PycharmProjects/mali/tools/webhook_test.py", line 238, in gym_or_invalid
    payload["message"][prm], payload["message"][prm2] = check_gym(raw_input())
TypeError: 'NoneType' object is not iterable
```